spec/controllers/users/verify_personal_key_controller_spec.rb

```diff
@@ -109,22 +109,49 @@
 
         expect(subject.reactivate_account_session.validated_personal_key?).to eq(true)
       end
+
+      it 'tracks irs attempts api for relevant users' do
+        stub_attempts_tracker
+
+        expect(@irs_attempts_api_tracker).to receive(:track_event).with(
+          :personal_key_reactivation_submitted,
+          failure_reason: {},
+          success: true,
+        ).once
+
+        post :create, params: { personal_key: profiles.first.personal_key }
+
+        expect(subject.reactivate_account_session.validated_personal_key?).to eq(true)
+      end
     end
 
     context 'with an invalid form' do
       let(:bad_key) { 'baaad' }
 
-      before do
+      it 'sets an error in the flash' do
         post :create, params: { personal_key: bad_key }
-      end
 
-      it 'sets an error in the flash' do
         expect(flash[:error]).to eq(error_text)
       end
 
       it 'redirects to form' do
+        post :create, params: { personal_key: bad_key }
         expect(response).to redirect_to(verify_personal_key_url)
       end
+
+      it 'tracks irs attempts api for relevant users' do
+        stub_attempts_tracker
+
+        expect(@irs_attempts_api_tracker).to receive(:track_event).with(
+          :personal_key_reactivation_submitted,
+          failure_reason: personal_key_error,
+          success: false,
+        ).once
+
+        allow_any_instance_of(VerifyPersonalKeyForm).to receive(:submit).and_return(response_bad)
```

Comment:
I think we could do this and stub the specific instance instead of any instance:
```suggestion
        allow(controller.personal_key_form).to receive(:submit).and_return(response_bad)
```

Reply:
Tried that, its a private form and unless we want to send to access the private method, or move it out from it being a private method. this initial approach doesnt work. Going to see an alternative approach.

Reply:
In the grand scheme of things, it's fine, it's not critical to change the style of this stub, it's fine to leave as-is too